plugins/languages/treesitter/treesitter.nix

```
@@ -6,210 +6,264 @@
-                scope_incremental = cfg.incrementalSelection.keymaps.scopeIncremental;
-                node_decremental = cfg.incrementalSelection.keymaps.nodeDecremental;
-              };
-            }
-          else
-            null;
-
-        indent = if cfg.indent then { enable = true; } else null;
-
-        ensure_installed = if cfg.nixGrammars then [ ] else cfg.ensureInstalled;
-        ignore_install = cfg.ignoreInstall;
-        parser_install_dir = cfg.parserInstallDir;
-      } // cfg.moduleConfig;
-    in
-    mkIf cfg.enable {
-      extraConfigLua =
-        (optionalString (cfg.parserInstallDir != null) ''
-          vim.opt.runtimepath:append("${cfg.parserInstallDir}")
-        '')
-        + ''
-          require('nvim-treesitter.configs').setup(${helpers.toLuaObject tsOptions})
-        ''
-        + (optionalString (cfg.languageRegister != { }) ''
-          __parserFiletypeMappings = ${helpers.toLuaObject cfg.languageRegister}
-
-          for parser_name, ft in pairs(__parserFiletypeMappings) do
-            require('vim.treesitter.language').register(parser_name, ft)
-          end
-        '');
-
-      extraFiles = mkIf cfg.nixvimInjections {
-        "queries/nix/injections.scm" = ''
-          ;; extends
-
-          (binding
-            attrpath: (attrpath (identifier) @_path)
-            expression: [
-              (string_expression (string_fragment) @lua)
-              (indented_string_expression (string_fragment) @lua)
-            ]
-            (#match? @_path "^extraConfigLua(Pre|Post)?$"))
-
-          (binding
-            attrpath: (attrpath (identifier) @_path)
-            expression: [
-              (string_expression (string_fragment) @vim)
-              (indented_string_expression (string_fragment) @vim)
-            ]
-            (#match? @_path "^extraConfigVim(Pre|Post)?$"))
+    grammarPackages = mkOption {
+      type = with types; listOf package;
+      default = config.plugins.treesitter.package.passthru.allGrammars;
+      defaultText = literalExpression "pkgs.vimPlugins.nvim-treesitter.passthru.allGrammars";
+      description = "Grammar packages to install";
+    };
+
+    languageRegister = mkOption {
+      type = with types; attrsOf (coercedTo str toList (listOf str));
+      default = { };
+      description = ''
+        This is a wrapping of the `vim.treesitter.language.register` function.
+        Register specific parsers to one or several filetypes.
+        The keys are the parser names and the values are either one or several filetypes.
+      '';
+    };
+
+    nixGrammars = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Whether to automatically install grammar packages.";
+    };
+
+    nixvimInjections = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''Whether to enable Nixvim injections, such as recognising extraConfigLua as lua.'';
+    };
+  };
+
+  callSetup = false;
```

#1823 added an `installPackage` option, we should disable that because the package is installed manually.
```suggestion
  callSetup = false;
  installPackage = false;
```